### PR TITLE
feature/TIN-17

### PR DESCRIPTION
**Pull request by tsenovi**

### CODE REVIEW

File: src/main/java/com/volasoftware/tinder/controllers/VerificationController.java [hunk 1]
```diff
@@ -46,4 +51,24 @@ public ResponseEntity<?> verify(@ApiParam(value = "Verify email")
...
+        value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation!"),
+            @ApiResponse(responseCode = "404", description = "Account not found!"),
+            @ApiResponse(responseCode = "406", description = "Account already verified!")
+        }
+    )
+    @PostMapping(value = "/resend-verification-email")
+    public ResponseEntity<?> resendVerificationMail(
+        @ApiParam(value = "Account Email", required = true)
+        @Valid
```
Comment (Kr1s4an) [inline src/main/java/com/volasoftware/tinder/controllers/VerificationController.java:66]: Are you validating anything for the purpose of this anotation?

File: src/main/java/com/volasoftware/tinder/services/implementations/VerificationTokenServiceImpl.java [hunk 2]
```diff
@@ -53,6 +54,26 @@ public Account verifyToken(String token) {
...
+        VerificationToken verificationToken = optionalVerificationToken.get();
+        String uuidToken = UUID.randomUUID().toString();
+        LocalDateTime currentTime = LocalDateTime.now();
+        verificationToken.setToken(uuidToken);
+        verificationToken.setLastModifiedDate(currentTime);
+        verificationToken.setExpiresAt(currentTime.plusDays(SecurityConstant.TOKEN_EXPIRATION_DAYS)
+        );
+
+        return verificationTokenRepository.save(verificationToken);
+    }
```
Comment (Kr1s4an) [inline src/main/java/com/volasoftware/tinder/services/implementations/VerificationTokenServiceImpl.java]: What is happening with the old token?

File: src/main/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -155,6 +156,19 @@ public AccountDto recoverPassword(Principal principal) {
         return accountMapper.accountToAccountDto(savedAccount);
     }
 
+    @Override
+    public EmailDto reverify(EmailDto emailDto) {
```
Comment (nadezhdaVacheva) [inline src/main/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImpl.java]: Why do methods like this and regenerateToken exist? Are there methods that do the same thing? If there are, it's appropriate to reuse them. If they do not exist, then you need to change the method's names

**Review by nadezhdaVacheva — CHANGES_REQUESTED** [review]

File: src/main/java/com/volasoftware/tinder/controllers/VerificationController.java (hunk 1 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/controllers/VerificationController.java:66]: The user should be able to enter a valid email address.

File: src/main/java/com/volasoftware/tinder/services/implementations/VerificationTokenServiceImpl.java (hunk 2 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/services/implementations/VerificationTokenServiceImpl.java]: If the account is not verified and there is already token, the record will be updated in the database.

File: src/main/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImpl.java (hunk 3 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImpl.java]: The methods are not doing the same thing, so they were renamed.

**Review by nadezhdaVacheva — APPROVED** [review]

**Pull request merged**